lib/activerecord-multi-tenant/query_rewriter.rb

```diff
@@ -375,11 +375,20 @@ def relations_from_node_join(node_join)
 ActiveRecord::QueryMethods.prepend(MultiTenant::QueryMethodsExtensions)
 
 module MultiTenantFindBy
-  def cached_find_by_statement(key, &block)
-    return super unless respond_to?(:scoped_by_tenant?) && scoped_by_tenant?
+  if ActiveRecord::VERSION::MAJOR >= 7 && ActiveRecord::VERSION::MINOR >= 2
+    def cached_find_by_statement(connection, key, &block)
+      return super unless respond_to?(:scoped_by_tenant?) && scoped_by_tenant?
 
-    key = Array.wrap(key) + [MultiTenant.current_tenant_id.to_s]
-    super
+      key = Array.wrap(key) + [MultiTenant.current_tenant_id.to_s]
+      super(connection, key, &block)
+    end
+  else
+    def cached_find_by_statement(key, &block)
+      return super unless respond_to?(:scoped_by_tenant?) && scoped_by_tenant?
+
+      key = Array.wrap(key) + [MultiTenant.current_tenant_id.to_s]
+      super(key, &block)
```

Comment:
```suggestion
      super(connection, Array.wrap(key) + [MultiTenant.current_tenant_id.to_s], &block)
    end
  else
    def cached_find_by_statement(key, &block)
      return super unless respond_to?(:scoped_by_tenant?) && scoped_by_tenant?

      super(Array.wrap(key) + [MultiTenant.current_tenant_id.to_s], &block)
```
not sure if super with implicit args picks up on key being rebound, maybe inlining expression gets past static check